src/wire/ipv6ext.rs

```diff
@@ -0,0 +1,246 @@
+    pub fn parse<T>(packet: &'a Packet<&'a T>) -> Result<Repr<'a>> where T: AsRef<[u8]> + ?Sized {
+        match packet.option_type() {
+            OptionType::Pad1 => {
+                Ok(Repr::Pad1 {
+                    ident: packet.option_type(),
+                })
+            }
+            OptionType::PadN => {
+                Ok(Repr::PadN {
+                    ident: packet.option_type(),
+                    length: packet.option_data_length(),
+                    data: packet.option_data(),
+                })
+            }
+            _ => Err(Error::Unrecognized),
+        }
+    }
+
+    /// Return the length of a header that will be emitted from this high-level representation.
+    pub fn buffer_len(&self) -> usize {
+        // This function is not strictly necessary, but it can make client code more readable.
+        match self {
+            &Repr::Pad1{..} => {
+                1
+            }
+            &Repr::PadN{length, ..} => {
+                (length as usize) + field::DATA.start
+            }
+
+            &Repr::__Nonexhaustive => unreachable!()
+        }
+    }
+
+    /// Emit a high-level representation into an IPv6 Extension Header Option packet.
+    pub fn emit<T: AsRef<[u8]> + AsMut<[u8]>>(&self, packet: &mut Packet<T>) {
+        match self {
+            &Repr::Pad1{ident} => {
+                packet.set_option_type(ident);
+            }
+            &Repr::PadN{ident, length, data} => {
+                packet.set_option_type(ident);
+                packet.set_option_data(data, length);
+            }
+
+            &Repr::__Nonexhaustive => unreachable!()
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_option_deconstruct() {
+        // one octet of padding
+        let bytes:  [u8; 1] = [0x0];
+        let packet = Packet::new(&bytes);
+        assert_eq!(packet.option_type(), OptionType::Pad1);
+
+        // two octets of padding
+        let bytes:  [u8; 2] = [0x1, 0x0];
+        let packet = Packet::new(&bytes);
+        assert_eq!(packet.option_type(), OptionType::PadN);
+        assert_eq!(packet.option_data_length(), 0);
+
+        // three octets of padding
+        let bytes:  [u8; 3] = [0x1, 0x1, 0x0];
+        let packet = Packet::new(&bytes);
+        assert_eq!(packet.option_type(), OptionType::PadN);
+        assert_eq!(packet.option_data_length(), 1);
+        assert_eq!(packet.option_data(), &[0]);
+
+        // extra bytes in buffer
+        let bytes:  [u8; 10] = [0x1, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xff];
+        let packet = Packet::new(&bytes);
+        assert_eq!(packet.option_type(), OptionType::PadN);
+        assert_eq!(packet.option_data_length(), 7);
+        assert_eq!(packet.option_data(), &[0, 0, 0, 0, 0, 0, 0]);
+
+        // unrecognized option actions
+        let bytes:  [u8; 1] = [0xff];
+        let packet = Packet::new(&bytes);
```

Comment:
Just FYI `Packet::new` bytes here should work, but `Packet::new_checked` here should fail.

Reply:
added test